### ISSUE TITLE
exception from the observatory backend

### ISSUE THREAD

**Issue by devoncarew** [dart-lang/sdk#23742]
When sending the observatory a command via the json-rpc protocol, it throws an exception (an NPE trying to get the keys property of a map) if the `params` field is not sent in. This is for method calls that do not take args, like `getVM`.

```
NoSuchMethodError: method not found: 'keys'
Receiver: null
Arguments: []
Stack Trace:
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
#2      _HashVMBase&MapMixin.addAll (dart:collection/maps.dart:58)
#3      Message.Message.fromJsonRpc (message.dart:39:12)
#4      WebSocketClient.onWebSocketMessage (server.dart:32:29)
#5      WebSocketClient.WebSocketClient.<anonymous closure> (server.dart:14:32)
```

@johnmccutchan

**Comment by johnmccutchan** [comment]
I'm pretty sure I fixed this last week, but in the meantime, send an empty params map

{
"method": "getVM"
"params": {}
}

**Comment by devoncarew** [comment]
Yup, I switched to sending that, I just want to give you a heads up :)

**Comment by devoncarew** [comment]
This is for 1.11.0-dev.5.2.

**Comment by johnmccutchan** [comment]
This was fixed in 983395f5f3d9d0714bf0e6ed47e58ac16ca47341
